component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java

```diff
@@ -1349,10 +1350,44 @@ public Response getSpaceExternalInvitations(@Context UriInfo uriInfo,
     if (space == null ||  (! spaceService.isManager(space, authenticatedUser) && ! spaceService.isSuperManager(authenticatedUser))) {
       throw new WebApplicationException(Response.Status.UNAUTHORIZED);
     }
-    List<SpaceExternalInvitationEntity> spaceExternalInvitations = spaceService.findSpaceExternalInvitationsBySpaceId(id);
+    List<SpaceExternalInvitation> spaceExternalInvitations = spaceService.findSpaceExternalInvitationsBySpaceId(id);
     return EntityBuilder.getResponseBuilder(spaceExternalInvitations, uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK).build();
   }
 
+
+  /**
+   * {@inheritDoc}
+   */
+  @DELETE
+  @Path("{id}/externalInvitations/{invitationId}")
+  @RolesAllowed("users")
+  @ApiOperation(value = "Delete a specific external invitations from a specific space",
+          httpMethod = "GET",
+          response = Response.class,
+          notes = "This Delete a specific external invitations from a specific space if the authenticated user is a member or manager of the space or a spaces super manager.")
+  @ApiResponses(value = {
+          @ApiResponse(code = 200, message = "Request fulfilled"),
+          @ApiResponse(code = 500, message = "Internal server error"),
+          @ApiResponse(code = 400, message = "Invalid query input")})
+  public Response declineExternalInvitations(@Context UriInfo uriInfo,
+                                             @ApiParam(value = "Space id", required = true) @PathParam("id") String id,
+                                             @ApiParam(value = "invitation id", required = true) @PathParam("invitationId") String invitationId) {
+
+    String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
+    //
+    Space space = spaceService.getSpaceById(id);
+    if (space == null ||  (! spaceService.isManager(space, authenticatedUser) && ! spaceService.isSuperManager(authenticatedUser))) {
+      throw new WebApplicationException(Response.Status.UNAUTHORIZED);
```

Comment:
Could you change the error from unauthorized (401)  to not found (404) to hide the fact that the token is valid.

Reply:
unauthorized (401) here means that only the only space manager or super use are authorized to remove the invitation